File: app/src/main/java/net/apocalypt/modernmovies/model/ListTypeConverters.java
package net.apocalypt.modernmovies.model;

import androidx.room.TypeConverter;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.List;

public class ListTypeConverters {

    @TypeConverter
    public List<Integer> fromString(String value) {
        Type listType = new TypeToken<List<String>>() {}.getType();
        return new Gson().fromJson(value, listType);
    }

    @TypeConverter
    public String fromArrayListInt(List<Integer> list) {
        Gson gson = new Gson();
        String json = gson.toJson(list);
        return json;
    }

}

File: app/src/main/java/net/apocalypt/modernmovies/model/ListTypeConverters.kt
package net.apocalypt.modernmovies.model

import androidx.room.TypeConverter
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

import java.lang.reflect.Type

class ListTypeConverters {

    @TypeConverter
    fun fromString(value: String): List<Int>? {
        val listType = object : TypeToken<List<String>>() {

        }.type
        return Gson().fromJson<List<Int>>(value, listType)
    }

    @TypeConverter
    fun fromArrayListInt(list: List<Int>): String {
        val gson = Gson()
        return gson.toJson(list)
    }

}
